pages/k8s/certs-and-trust.md

```diff
@@ -0,0 +1,247 @@
+will have to be configured as an intermediary CA with a chain of trust up to a
+globally trusted CA (such as [Comodo][] or [DigiCert][]).
+
+The certificate information will also be included in the **Kubernetes** `config`
+file that **CDK** generates to be used by [kubelet][], so that communications
+between the local machine and Kubernetes will be secured.
+
+## Server certificates
+
+Each service that will be connected to will need a [server certificate][]
+identifying and validating it to any clients that wish to connect.
+
+The primary address at which the service can be reached will be its [common
+name][CN]; ideally, this will be a fully-qualified domain name (FQDN) which
+will not change, but for internal service communication, it is often just the
+[ingress address][network primitives] for the unit. Any additional names or
+addresses by which the service can be reached will be its [subject alternative
+names (SANs)][SANs].
+
+**CDK** will manage the server certificates automatically, including
+generating the certificate with the proper CN and SANs. However, the
+[kubernetes-master][] charm also supports an [`extra_sans`][] option which
+can be used to provide additional names to be added to the SANs list.
+
+## Client certificates
+
+In order to provide for two-way security, some services require that clients
+identify themselves via a [client certificate][]. These are more or less the
+same as server certificates, but are presented by a client to the service
+they are connecting to, so that the service can validate that client's identity.
+Client certificates can _only_ be used to identify a client and will be
+rejected by clients if presented by a service they are connecting to.
+
+# Certificate Authorities for CDK
+
+**CDK** can use a CA provided by any charm which [provides a tls-certificates
+endpoint][provides-tls]. The two current recommendations are
+[EasyRSA](#easyrsa) and [Vault](#vault).
+
+## EasyRSA
+
+By default, the **CDK** bundle includes the [EasyRSA charm][easyrsa-charm]. This
+is a relatively simple charm which uses [OpenVPN's easy-rsa][easy-rsa] to
+provide a CA and sign certificates. This is lightweight and works out
+of the box without any additional configuration, but it cannot act as an
+intermediary CA and does not support [HA][].
+
+## Vault
+
+For production systems, it is recommended to replace EasyRSA with the [Vault
+charm][vault-charm]. This uses [HashiCorp's Vault][vault] to provide either a
+root or intermediate CA. It can also be deployed HA, as well provide a secure
+secrets store which can be used to enable encryption-at-rest for **CDK**. However,
+it requires a database to store its data, and (depending on configuration)
+some manual steps will be required after deployment as well as after any reboot
+to unseal Vault so that the secrets, such as certificates and signing keys, can
+be accessed.
+
+### Deploying CDK with Vault as a root CA
+
```

Comment:
Ideally I would link to this from the 'Manual Install page', erm, when it is written.

Presumably it is also possible to replace EastRSA with Vault after a deploy, and I wonder if I shouldn't document that case as well.